hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -264,6 +251,146 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
+      LOG.info("Skipping drop, View does not exist: {}", identifier, e);
+      return false;
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop " + identifier, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to dropView", e);
+    }
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    try {
+      return listContents(namespace, TableType.VIRTUAL_VIEW.name(), icebergPredicate());
+    } catch (UnknownDBException e) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all views under namespace " + namespace, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listViews", e);
+    }
+  }
+
+  private List<TableIdentifier> listContents(
+      Namespace namespace, String tableType, Predicate<Table> tablePredicate)
+      throws TException, InterruptedException {
+    Preconditions.checkArgument(
+        isValidateNamespace(namespace), "Missing database in namespace: %s", namespace);
+    String database = namespace.level(0);
+    List<String> tableNames =
+        StringUtils.isNotEmpty(tableType)
+            ? clients.run(client -> client.getTables(database, "*", TableType.valueOf(tableType)))
+            : clients.run(client -> client.getAllTables(database));
+    List<Table> tableObjects =
+        clients.run(client -> client.getTableObjectsByName(database, tableNames));
+    List<TableIdentifier> tableIdentifiers =
+        tableObjects.stream()
+            .filter(tablePredicate)
+            .map(table -> TableIdentifier.of(namespace, table.getTableName()))
+            .collect(Collectors.toList());
+
+    LOG.debug(
+        "Listing of namespace: {} for table type {} resulted in the following: {}",
+        namespace,
+        tableType,
+        tableIdentifiers);
+    return tableIdentifiers;
+  }
+
+  private Predicate<Table> icebergPredicate() {
+    return table ->
+        table.getParameters() != null
+            && BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.equalsIgnoreCase(
+                table.getParameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP));
+  }
+
+  @Override
+  @SuppressWarnings("FormatStringAnnotation")
+  public void renameView(TableIdentifier from, TableIdentifier originalTo) {
+
+    if (!isValidIdentifier(from)) {
+      throw new NoSuchViewException("Invalid identifier: %s", from);
+    }
+
+    if (!namespaceExists(originalTo.namespace())) {
+      throw new NoSuchNamespaceException(
+          "Cannot rename %s to %s. Namespace does not exist: %s",
+          from, originalTo, originalTo.namespace());
+    }
+
+    TableIdentifier to = removeCatalogName(originalTo);
```

Comment:
catalog name should be removed for `from` also?

Reply:
I followed the idea of [renameTable](https://github.com/apache/iceberg/blob/4e62b58f04aba4ccc2a2a846494b478d3b03f58f/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java#L229)